crates/sui-config/src/node.rs

```diff
@@ -560,11 +560,18 @@ pub struct AuthorityStorePruningConfig {
     /// number of epochs to keep the latest version of transactions and effects for
     #[serde(skip_serializing_if = "Option::is_none")]
     pub num_epochs_to_retain_for_checkpoints: Option<u64>,
-    /// enables pruner to prune no longer needed object tombstones. We don't serialize it if it is the default value, false.
-    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
+    /// enables pruner to prune no longer needed object tombstones. We don't serialize it if it is the default value, true.
+    #[serde(
+        default = "default_enable_pruning_tombstones",
+        skip_serializing_if = "Clone::clone"
+    )]
     pub enable_pruning_tombstones: bool,
 }
 
+fn default_enable_pruning_tombstones() -> bool {
+    true
```

Comment:
We don't have pruning enabled by default in our configs. So it might come as a surprise to some. I may suggest we keep this default false. @phoenix-o thoughts?

Reply:
tombstone pruning is enabled only if the objects pruning itself is enabled.
nit: so if that's a temporary setting, maybe rename it into `killswitch_tombstone_pruning` and set it to false by default?

Reply:
Sounds good. Seems to be a bigger change and may cause more conflict with #14851 . Maybe I can wait until that PR is in.

Reply:
PR is ready for review!